src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputExtension.java

```diff
@@ -0,0 +1,33 @@
+package org.jenkinsci.plugins.workflow.support.steps.input;
+
+import hudson.ExtensionPoint;
+import hudson.model.Run;
+
+public interface InputExtension extends ExtensionPoint {
+    /**
+     * Callback this when an inputStep is executing.
+     * Just for notification, will ignore all exception and log them.
+     * @param inputStep Instance of InputStep
+     * @param run Current building run instance
+     * @param userID Id of a user who triggered the action
+     * @param notifyEvent Event type of notification
+     */
+    void notifyInput(InputStep inputStep, Run run, String userID, NotifyEvent notifyEvent);
+
+    /**
+     * Should be unique, for distinguish different extension.
+     * @return the name of current extension instance
+     */
+    default String getName() {
+        return this.getClass().getSimpleName();
+    }
+
+    enum NotifyEvent {
```

Comment:
Nit: Personally I'd prefer a name more like `InputEvent` with variants `STARTED`, `APPROVED`, and `REJECTED`, but other reviewers might disagree with me.

Reply:
IMO someone how wants to approve or reject that is a specific use case. Because if the button of abort was pressed, then the whole pipeline will be aborted. But the rejection sounds like the process can be continued.
So if anyone wants to use this DSL `input` to reject a stage, then he needs to provide a parameter for this.

I agree with your ideas about using `InputEvent` and `STARTED`.